components/shared_code/src/shared/model/metric.py

```diff
@@ -159,9 +160,14 @@ def summarize(self, measurements: list[Measurement], **kwargs) -> dict:
         summary["scale"] = self.scale()
         summary["status"] = self.status(latest_measurement)
         summary["status_start"] = latest_measurement.status_start() if latest_measurement else None
-        summary["latest_measurement"] = latest_measurement
+        summary["latest_measurement"] = latest_measurement.summarize_latest() if latest_measurement else None
         summary["recent_measurements"] = [measurement.summarize() for measurement in measurements]
         if latest_measurement:
             summary["issue_status"] = self.issue_statuses(latest_measurement)
         summary.update(kwargs)
         return summary
+
+    def source_parameter_hash(self) -> str:  # pragma: no feature-test-cover
+        """Return a hash of the source parameters that can be used to check for changed parameters."""
+        source_parameters = [source.get("parameters") for source in self.sources]
```

Comment:
Is the assumption that these objects are always listed in the same order, safe?

Reply:
Good question. I think the order may change if the user reorders sources. I'll add sorting by source UUID before generating the hash.